test/Sync.t.sol

```diff
@@ -88,6 +88,35 @@ contract SyncTest is Test, Deployers, GasSnapshot {
         assertEq(manager.getReserves(), balanceCurrency2);
     }
 
+    function test_settle_payOnBehalf(address recipient, uint256 amount) public {
+        vm.assume(recipient != address(router));
+        amount = bound(amount, 1, uint256(int256(type(int128).max)));
+        MockERC20(Currency.unwrap(currency2)).approve(address(router), type(uint256).max);
+
+        Actions[] memory actions = new Actions[](6);
+        bytes[] memory params = new bytes[](6);
+
+        actions[0] = Actions.SYNC;
+        params[0] = abi.encode(currency2);
+
+        actions[1] = Actions.TRANSFER_FROM;
+        params[1] = abi.encode(currency2, address(this), address(manager), amount);
+
+        actions[2] = Actions.SETTLE_FOR;
+        params[2] = abi.encode(recipient);
+
+        actions[3] = Actions.ASSERT_DELTA_EQUALS;
+        params[3] = abi.encode(currency2, recipient, amount);
+
+        actions[4] = Actions.ASSERT_DELTA_EQUALS;
+        params[4] = abi.encode(currency2, address(router), 0);
+
+        actions[5] = Actions.PRANK_TAKE_FROM;
```

Comment:
I'm a little confused what this is testing.. 

We are encoding here (currency2, recipient, address(this), amount)

but then inside `_prankTakeFrom` we are decoding (currency, from, recipient, amount)

Can you add another assert after this call for what you expect to happen?

Reply:
with settleFor we are settling on behalf of a third party. In order to be able to close out the delta, I'm pranking from the address that is the recipient in the settleFor call in order to successfully close out the deltas. Thus, the functionality is also implicitly tested, because the deltas are closed

Reply:
yeah +1 to what sara said. The command decodes
```
(Currency currency, address from, address recipient, uint256 amount) =
            abi.decode(params, (Currency, address, address, uint256));
```

But this call has the recipient second not third. Please can you check it and add checks 🙏

Reply:
no this is correct, the recipient is the settleFor recipient, the account that receives the positive delta during the settleFor call. So that's the account we need to prank (from). The recipient in take is the recipient of the funds

Reply:
I think I see whats happening and IMO its just confusing because the params are the same names for different contexts. I think what you are testing is.. apply a positive delta for recipient, then prank that recipient and send it to somewhere else.

I think what I would prefer to see tested is pranking another address such that there is a negative delta open on that address then someone else transfers their own funds, and calls settleFor with recipient to pay off their negative delta